src/sage/algebras/free_algebra.py

```diff
@@ -774,6 +804,23 @@ def gens(self):
         """
         return tuple(self.gen(i) for i in range(self.__ngens))
 
+    def degree_on_basis(self, m):
+        r"""
+        Return the degree of the basis element indexed by ``m``.
+
+        EXAMPLES::
+
+            sage: A.<a, b> = FreeAlgebra(QQ, degrees=(1, -1))
+            sage: m = A.basis().keys()[42]
+            sage: m
+            a*b*a*b^2
+            sage: A.degree_on_basis(m)
+            -1
+            sage: (a*b*a*b^2).degree()
+            -1
+        """
+        return sum(self._degrees[g] * e for g, e in m)
```

Comment:
```suggestion
        return ZZ.sum(self._degrees[g] * e for g, e in m)
```
So everything is an integer; in particular the degree of the constants. It would be good to add a doctest for this.